File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

long long solution(long long n) 
{
    long long root = sqrt(n);
    if (root * root == n) 
    {
        // n이 완전 제곱수일 경우 다음 완전 제곱수를 반환
        return (root + 1) * (root + 1);
    }
    // n이 완전 제곱수가 아닐 경우 -1 반환
    return -1;
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cs
public class Solution 
{
    public long solution(long n) 
    {
        long s = 1;
        while(s * s < n)
        {
            s++;
        }
    
        if(s * s == n)
        {
            return (s+1) * (s+1);
        }
        else
        {
            return -1;
        }
    }
}